### ISSUE TITLE
Save as dialog sometimes get captured

### ISSUE THREAD

**Issue by pepa65** [phw/peek#90]
Which I don't think anybody wants. Not sure if it's just me... I've just implemented a little sleep here, which helps. There might be a better solution.

**Comment by phw** [comment]
I have not yet experienced this myself so far, but adding a small amount of
delay does surely no harm, so I will do it.

I would still like to know what's happening, though. Actually the dialog
should show up only after recording has finished, but there was a bug with
this before version 0.9. Which version are you using, 0.8, 0.9 or latest
daily?

If latest daily, are you by any chance running on Gnome 3 (because the
daily will use Gnome's built-in recording)

Am 26.02.2017 10:18 vorm. schrieb "PePa" <notifications@github.com>:

**Comment by pepa65** [comment]
Just what's current on Github's master branch. I'm running Ubuntu MATE 16.04.

**Comment by phw** [comment]
One last question: When you start Peek from command line, what screen recorder backend does it show? I assume ffmpeg, it should output something like:

    Using screen recorder backend ffmpeg

**Comment by pepa65** [comment]
It does output that when you start Peek from the commandline. It's either ffmpeg or avconv.

**Comment by phw** [comment]
I think I could fix it properly without the timeout. Actually the showing the file chooser and sending the stop signal happened in parallel, so if stopping the recording process took to long it actually could happen that the file chooser itself got recorded (I still could not reproduce it on my machine, though). This is fixed now and the file chooser is only shown after the recording process stopped.

Also while debugging this I realized that running the post processing on the file was not started until the file chooser closed (I broke this with a recent update). Now the order of processing is fixed again and the post processing happens in parallel to the user choosing the save file.

Thanks for reporting this and being patient with all my questions :)
